### PR TITLE
Create a single tmp dir per test session

### PR DESCRIPTION
**Pull request by delucchi-cmu**
Previous behavior was to create a new tmp directory for each test case that required writing to the cloud bucket. That adds a lot of overhead, as removing the directory is a slow, asynchronous call. We have an exponential backoff and retry to delete, but there were around 600 temp directories that were not cleaned up due to failed attempts.

This change creates a single temp directory in the cloud bucket for the whole execution, and cleans it up at the end of the testing session. I increase the number of retries, to get one more chance to successfully delete before giving up.

Because we only have ONE exponential backoff/retry cycle for the whole test session, we trim about 15-30 seconds off the total execution time

In addition, since this changed so many test files, I figured it was an ok time to rename the overly-long `example_cloud_storage_options` fixture. Also cleans up the session-level fixtures.

### CODE REVIEW

File: tests/conftest.py [hunk 1]
```diff
@@ -65,35 +65,52 @@ def small_sky_parts_dir_local(local_data_dir):
...
-def small_sky_index_dir_cloud(example_cloud_path):
-    return os.path.join(example_cloud_path, "data", "small_sky_object_index")
+def small_sky_margin_dir_cloud(cloud_path):
+    return os.path.join(cloud_path, "data", "small_sky_order1_margin")
+
+
+@pytest.fixture(scope="session", name="tmp_dir_cloud")
+def tmp_dir_cloud(cloud_path, storage_options):
+    """Create a single client for use by all unit test cases."""
+    # client = Client()
```
Comment (jeremykubica) [inline tests/conftest.py:100]: Is this line still needed?

File: tests/conftest.py [hunk 2]
```diff
@@ -65,35 +65,52 @@ def small_sky_parts_dir_local(local_data_dir):
...
+    tmp = TempCloudDirectory(
+        os.path.join(cloud_path, "tmp"),
+        method_name="full_test",
+        storage_options=storage_options,
+    )
+    yield tmp.open()
```
Comment (jeremykubica) [inline tests/conftest.py:106]: Will this do a new open each time `tmp_dir_cloud` is called?

I was expecting something like:
```
path = tmp.open()
yield path
```

but I'm not actually sure how this works in Python.
```

**Review by jeremykubica — COMMENTED** [review]
Looks like a nice change in terms of efficiency. I left two questions for my own understanding.

File: tests/conftest.py (hunk 2 above)
Comment (delucchi-cmu) [reply tests/conftest.py:106]: Short answer: no.

Longer answer: some kinda python magic makes sure this is only called once, and by using `yield`, we can clean up whatever state we need to after the `yield` call. There's some caching happening, and I don't fully understand.

But in testing, I did confirm that it's all using the same opened directory.

**Review by jeremykubica — APPROVED** [review]

**Pull request merged**